src/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs

```diff
@@ -393,43 +391,35 @@ internal static HttpParseResult GetQuotedPairLength(string input, int startIndex
                 // If we support nested expressions and we find an open-char, then parse the nested expressions.
                 if (supportsNesting && (input[current] == openChar))
                 {
-                    nestedCount++;
-                    try
+                    // Check if we exceeded the number of nested calls.
+                    if (nestedCount > maxNestedCount)
```

Comment:
Nit: as long as you're touching this code, mind changing `maxNestedCount` to `MaxNestedCount`?

Reply:
done